libraries/resource_sensu_metric_dirsize.rb

```diff
@@ -0,0 +1,17 @@
+require "chef/resource"
+
+class Chef
+  class Resource
+    class SensuPluginMetricDirsize < Chef::Resource::SensuPluginCheckGeneric
```

Comment:
This should derive from `SensuPluginMetricGeneric` (which you added :-) ) and not from `SensuPluginCheckGeneric`